packages/atomic/src/components/atomic-refine-modal/atomic-refine-modal.tsx

```diff
@@ -35,15 +36,50 @@ export class AtomicRefineModal implements InitializableComponent {
   @State() public error!: Error;
 
   @Prop({reflect: true, mutable: true}) enabled!: boolean;
+  @Watch('enabled')
+  watchEnabled(enabled: boolean) {
+    const modalOpenedClass = 'atomic-modal-opened';
+
+    if (enabled) {
+      document.body.classList.add(modalOpenedClass);
+      this.duplicateFacetElements();
+      return;
+    }
+
+    document.body.classList.remove(modalOpenedClass);
+    this.flushFacetElements();
+  }
 
   public initialize() {
     this.breadcrumbManager = buildBreadcrumbManager(this.bindings.engine);
     this.querySummary = buildQuerySummary(this.bindings.engine);
   }
 
+  private duplicateFacetElements() {
+    const divSlot = document.createElement('div');
+    divSlot.setAttribute('slot', 'facets');
+    // TODO: order facets with facet manager
+    this.bindings.store.get('facetElements').forEach((facetElement) => {
+      const clone = facetElement.cloneNode(false) as HTMLElement;
+      clone.style.marginBottom =
+        'var(--atomic-refine-modal-facet-margin, 20px)';
+      clone.setAttribute('is-collapsed', 'true');
+      divSlot.append(clone);
+    });
+
+    this.host.append(divSlot);
+  }
+
+  private flushFacetElements() {
+    this.host.querySelector('div[slot="facets"]')?.remove();
```

Comment:
Can the modal contents be managed using JSX to avoid DOM queries?

Reply:
JSX will affect the content inside it's shadow dom. The facets are slotted, but they're not inside the shadow dom.

Reply:
Ok, thanks for the clarifications,